firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -1161,6 +1161,23 @@ class BrowserCoordinator: BaseCoordinator,
         router.push(webviewViewController)
     }
 
+    func popToBVC() {
+        guard let currentViewController = router.navigationController.topViewController else { return }
+
+        // Check if the current view controller is the shortcuts library, and if so, don't record closed telemetry when
+        // programatically popping back to BVC
+        (currentViewController as? ShortcutsLibraryViewController)?.recordTelemetryOnDisappear = false
```

Comment:
I am less of a fan that the coordinator handles side effects and business logic of the stories/shortcuts librariy view controller. I think this is more the responsibilities of the view controllers